cpp/src/io/parquet/reader_impl.cu

```diff
@@ -49,20 +49,78 @@ constexpr uint32_t PARQUET_COLUMN_BUFFER_SCHEMA_MASK          = (0xffffff);
 constexpr uint32_t PARQUET_COLUMN_BUFFER_FLAG_LIST_TERMINATED = (1 << 24);
 
 namespace {
+
+parquet::ConvertedType logical_type_to_converted_type(parquet::LogicalType const &logical)
+{
+  if (logical.__isset.STRING) {
+    return parquet::UTF8;
+  } else if (logical.__isset.MAP) {
+    return parquet::MAP;
+  } else if (logical.__isset.LIST) {
+    return parquet::LIST;
+  } else if (logical.__isset.ENUM) {
+    return parquet::ENUM;
+  } else if (logical.__isset.DECIMAL) {
+    return parquet::DECIMAL;  // TODO set decimal values
+  } else if (logical.__isset.DATE) {
+    return parquet::DATE;
+  } else if (logical.__isset.TIME) {
+    if (logical.TIME.unit.__isset.MILLIS)
+      return parquet::TIME_MILLIS;
+    else if (logical.TIME.unit.__isset.MICROS)
+      return parquet::TIME_MICROS;
+  } else if (logical.__isset.TIMESTAMP) {
+    if (logical.TIMESTAMP.unit.__isset.MILLIS)
+      return parquet::TIMESTAMP_MILLIS;
+    else if (logical.TIMESTAMP.unit.__isset.MICROS)
+      return parquet::TIMESTAMP_MICROS;
+  } else if (logical.__isset.INTEGER) {
+    switch (logical.INTEGER.bitWidth) {
+      case 8: return logical.INTEGER.isSigned ? INT_8 : UINT_8;
+      case 16: return logical.INTEGER.isSigned ? INT_16 : UINT_16;
+      case 32: return logical.INTEGER.isSigned ? INT_32 : UINT_32;
+      case 64: return logical.INTEGER.isSigned ? INT_64 : UINT_64;
+      default: break;
+    }
+  } else if (logical.__isset.UNKNOWN) {
+    return parquet::NA;
+  } else if (logical.__isset.JSON) {
+    return parquet::JSON;
+  } else if (logical.__isset.BSON) {
+    return parquet::BSON;
+  }
+  return parquet::UNKNOWN;
+}
+
 /**
  * @brief Function that translates Parquet datatype to cuDF type enum
  */
 type_id to_type_id(SchemaElement const &schema,
                    bool strings_to_categorical,
                    type_id timestamp_type_id)
 {
-  parquet::Type physical         = schema.type;
-  parquet::ConvertedType logical = schema.converted_type;
-  int32_t decimal_scale          = schema.decimal_scale;
+  parquet::Type physical                = schema.type;
+  parquet::ConvertedType converted_type = schema.converted_type;
+  int32_t decimal_scale                 = schema.decimal_scale;
+  std::cout << "physical=" << int(physical) << ",converted_type=" << int(converted_type)
```

Comment:
ignore the `std::cout` for now. they are for debug purposes. they will be removed finally.